.github/workflows/continuous-integration.yaml

```diff
@@ -0,0 +1,139 @@
+          path: ~/.npm
+          restore-keys: |
+            npm-
+      - name: Cache local cache
+        uses: actions/cache@v1
+        with:
+          key: eslint-${{ github.sha }}
+          path: .cache
+          restore-keys: |
+            eslint-
+      - name: Install dependencies
+        run: npm ci --ignore-scripts --no-audit --prefer-offline --progress=false
+      - name: Lint
+        run: |
+          # This is required to workaround a limitation of ESLint which uses
+          # file modification time in its cache, instead of file checksum.
+          #
+          # cspell:ignore iregex
+          find . \
+            -type d -name "node_modules" -prune -o \
+            -type f -iregex '.*.ts' \
+            -exec ./scripts/ci/set_mtime_to_md5.sh {} \;
+          npm run lint:js
+  lint-typescript:
+    name: Lint TypeScript declaration files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Cache npm dependencies
+        uses: actions/cache@v1
+        with:
+          key: npm-${{ hashFiles('**/package-lock.json') }}
+          path: ~/.npm
+          restore-keys: |
+            npm-
+      - name: Install dependencies
+        run: npm ci --ignore-scripts --no-audit --prefer-offline --progress=false
+      - name: Lint
+        run: |
+          npm run lint:ts
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Get npm cache directory
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
+      # The following two steps can be greatly simplified once
+      # https://github.com/actions/cache/issues/39 is fixed.
+      - if: runner.os != 'Windows'
+        name: Cache npm dependencies
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: |
+            npm-${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            npm-${{ runner.os }}-
+      - if: runner.os == 'Windows'
+        name: Cache npm dependencies
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: |
+            npm-${{ runner.os }}-${{ hashFiles('**\package-lock.json') }}
+          restore-keys: |
+            npm-${{ runner.os }}-
+      - name: Install dependencies
+        run: npm ci --ignore-scripts --no-audit --prefer-offline --progress=false
+      - # The following conditional can be removed once tests are fully
+        # migrated.
+        if: runner.os != 'Windows'
```

Comment:
Tap works, but Mocha script relies on a grep that I don't know how to make work on Windows. I ever work on Windows OS, maybe someone could give me a hand here? Otherwise we can just wait until the migration (if that's why I see both Tap and Mocha) is done.

Reply:
That seems fine. Would it make sense to add an extra step here that only runs on Windows, including the tap tests?

Reply:
I added a step that runs Tap tests in Windows: 09fb13068d7e33f0352cfdf06f0d6e2d920f0272. We can enable Mocha once the grep is no longer necessary.